### DISCUSSION TITLE
Test suite for node lib api

### DISCUSSION THREAD

**Discussion by Puellaquae** [nodejs/help#4985] [Q&A] [answered]
I want to know if I want to implement some polyfill for node lib api, such as fs, whether here has some test suite for node lib api which can test my implemention?

**Comment by bnoordhuis** [comment, marked as answer]
The `test/` directory in this repo?

**Reply by Puellaquae** [reply]
I have seen this directory, but most of it is about testing the internal implement of nodejs, not its lib api.

**Reply by bnoordhuis** [reply]
Some do but most test the expected behavior of the public API. A quick litmus test is whether it uses `internalBinding()` or anything from lib/internal.

Of the approximately 210 test/parallel/test-fs-* tests, only 10% test internal details.

**Reply by Puellaquae** [reply]
I see it. Thank you.
